### PR TITLE
Handle lldp rx timeout

### PR DESCRIPTION
**Pull request by olbjo**
Mark lldp peer as invalid on lldp timeout.

Needs to be verified in the setup problem was originally reported.
Does implementation work with the snmp integration?

### CODE REVIEW

**Review by pyhys — APPROVED** [review]
Great! Tested with Raspberry Pi with USB-to-network adapter

File: src/common/pf_lldp.c [hunk 1]
```diff
@@ -491,6 +513,7 @@ static void pf_lldp_receive_timeout (
    LOG_WARNING (PF_LLDP_LOG, "LLDP(%d): Receive timeout expired\n", __LINE__);
 
    pf_pdport_peer_lldp_timeout (net, p_port_data->port_num);
+   pf_lldp_peer_timeout(net, p_port_data->port_num);
```
Comment (fredrikdanielmoller) [inline src/common/pf_lldp.c:516]: Use clang format

File: src/common/pf_lldp.c [hunk 2]
```diff
@@ -466,6 +466,28 @@ static void pf_lldp_add_management (
...
 }
 
+/**
+ * Mark lldp peer data as invalid.
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param loc_port_num     In:    Local port number.
+ *                                Valid range: 1 .. PNET_MAX_PORT
+ */
+void pf_lldp_peer_timeout(pnet_t * net, int loc_port_num)
```
Comment (fredrikdanielmoller) [inline src/common/pf_lldp.c:476]:
```suggestion
void pf_lldp_invalidate_peer_info(pnet_t * net, int loc_port_num)
```

**Review by fredrikdanielmoller — CHANGES_REQUESTED** [review]

File: src/common/pf_lldp.c [hunk 3]
```diff
@@ -466,6 +466,28 @@ static void pf_lldp_add_management (
...
+
+   os_mutex_lock (net->lldp_mutex);
+   p_port_data->lldp.is_peer_info_received = false;
+   p_port_data->lldp.peer_info.chassis_id.is_valid = false;
+   p_port_data->lldp.peer_info.port_id.is_valid = false;
+   p_port_data->lldp.peer_info.port_description.is_valid = false;
+   p_port_data->lldp.peer_info.management_address.is_valid = false;
+   p_port_data->lldp.peer_info.port_delay.is_valid = false;
+   p_port_data->lldp.peer_info.phy_config.is_valid = false;
+   os_mutex_unlock (net->lldp_mutex);
```
Comment (fredrikdanielmoller) [inline src/common/pf_lldp.c:488]: Set `ttl `to zero, as that seems to mean that it is invalid.

File: src/common/pf_lldp.c (hunk 2 above)
Comment (olbjo) [reply src/common/pf_lldp.c]: Done

File: src/common/pf_lldp.c (hunk 3 above)
Comment (olbjo) [reply src/common/pf_lldp.c:489]: Done

File: src/common/pf_lldp.c (hunk 2 above)
Comment (olbjo) [reply src/common/pf_lldp.c]: Oups. didi not read comment properly will push another update..

Comment (olbjo) [reply src/common/pf_lldp.c]: Done

File: src/common/pf_lldp.c (hunk 1 above)
Comment (olbjo) [reply src/common/pf_lldp.c]: Done

**Review by fredrikdanielmoller — APPROVED** [review]

**Pull request merged**